diagnostic_updater/include/diagnostic_updater/diagnostic_updater.hpp

```diff
@@ -370,11 +370,45 @@ class Updater : public DiagnosticTaskVector
    * \param h Node handle from which to get the diagnostic_period
    * parameter.
    */
-  explicit Updater(rclcpp::Node::SharedPtr node = rclcpp::Node::make_shared("diagnostics_updater"))
-  : node_(node), node_name_(node_->get_name())
+  template<class NodeT>
+  explicit Updater(std::shared_ptr<NodeT> node, double period = 1.0)
+  : Updater(
+      node->get_node_base_interface(),
+      node->get_node_logging_interface(),
+      node->get_node_parameters_interface(),
+      node->get_node_timers_interface(),
+      node->get_node_topics_interface(),
+      period)
+  {}
+
+  Updater(
+    std::shared_ptr<rclcpp::node_interfaces::NodeBaseInterface> base_interface,
+    std::shared_ptr<rclcpp::node_interfaces::NodeLoggingInterface> logging_interface,
+    std::shared_ptr<rclcpp::node_interfaces::NodeParametersInterface> parameters_interface,
+    std::shared_ptr<rclcpp::node_interfaces::NodeTimersInterface> timers_interface,
+    std::shared_ptr<rclcpp::node_interfaces::NodeTopicsInterface> topics_interface,
+    double period = 1.0)
+  : verbose_(false),
+    publisher_(
+      rclcpp::create_publisher<diagnostic_msgs::msg::DiagnosticArray>(
+        topics_interface, "/diagnostics", 1)),
+    logger_(logging_interface->get_logger()),
+    node_name_(base_interface->get_name()),
+    warn_nohwid_done_(false)
   {
-    // @todo: how to deal with default node?
-    setup();
+    // TODO(karsten1987): Follow up on parameter once
```

Comment:
I understand the TODO here, but wouldn't it be better to do a one-time `parameter_get` so that those setting a static period can still use the parameter? At least until the callback is fixed?